### DISCUSSION TITLE
I want to change the position of the focus when I use shouldFocusError.

### DISCUSSION THREAD

**Discussion by shimotaroo** [react-hook-form/react-hook-form#7739] [Q&A] [answered]
## Preconditions

- one `<form></form>` in a page.
- These <input> exist on a page.
    - inputA (rule: required)
    - inputB (rule: required)
    - inputC (rule: required)
- inputB is hidden by default

## Event

default

- inputA
- inputC

after

- inputA
- inputB
- inputC

When I submit with iuputB and inputC empty, I want inputB to be the focus, but it focuses on inputC.

I think this happens because the React-Hook-Form `register` are added in the order they appear on the screen, is there a solution to this?

**Comment by Moshyfawn** [comment, marked as answer]
See #7523 #6982

**Reply by bluebill1049** [reply]
yes, `register` invoke order does matter. for custom behavior, I would recommend turning off `shouldFocus` and using `setFocus` to manually set field focus at the app level with custom logic.

**Reply by shimotaroo** [reply]
Thanks.
I could implement this by using setFocus.

**Reply by bluebill1049** [reply]
nice 👍

**Comment by shimotaroo** [comment]
Thank you for the other day.

I have written an article in Japanese on my blog about how to create your own focus logic using `setFocus`.

Here is the link to the article.

https://yutaro-blog.net/2022/02/22/react-hook-form-custom-focus/

**Reply by bluebill1049** [reply]
That's awesome! we have new `select` support as well now. 

```
setFocus('test', { shouldSelect: true })
```

(feature added by @Moshyfawn)

**Reply by Moshyfawn** [reply]
> (feature added by @Moshyfawn)

Ahaha, hand-delivered with the utmost care xD

Gonna checkout your article once I learn Japanese lol _(jk will find a good translation :P)_

**Reply by shimotaroo** [reply]
@bluebill1049 @Moshyfawn 

Thank you.

By the way, what does option `shouldSelect` mean?

**Reply by bluebill1049** [reply]
![Screen Shot 2022-02-25 at 7 30 35 pm](https://user-images.githubusercontent.com/10513364/155681841-c27e8ba9-b9bf-4022-9f53-1cacc3142761.png)

see above the whole text is selected.

**Reply by Moshyfawn** [reply]
[Release 7.27.0](https://github.com/react-hook-form/react-hook-form/releases/tag/v7.27.0)

**Reply by shimotaroo** [reply]
Thank you.
